[Xcode] WebCore fails to build for iOS when the macOS SDK is old

emw-apple opened the pull request:
#### 6a632b1cec5c756581934fbfee146933652486e0
<pre>
[Xcode] WebCore fails to build for iOS when the macOS SDK is old
<a href="https://bugs.webkit.org/show_bug.cgi?id=264848">https://bugs.webkit.org/show_bug.cgi?id=264848</a>
<a href="https://rdar.apple.com/118356369">rdar://118356369</a>

Reviewed by Alexey Proskuryakov.

Redefine USE_DAV1D with platform conditionals, so that iOS builds don&apos;t
depend on the version of the macOS SDK.

* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: <a href="https://commits.webkit.org/272438@main">https://commits.webkit.org/272438@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f17ac845922cc106c9958362e286c4e330cc2111

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31631 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/10310 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/33346 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/34120 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/28645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/32409 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12664 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7560 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28245 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31979 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8688 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/28228 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7488 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7650 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/28141 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35466 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28752 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28594 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/33774 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7741 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5752 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/31626 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9398 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7423 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8426 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/8250 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/PAL/Configurations/PAL.xcconfig:
```diff
@@ -57,11 +57,10 @@ USE_LIBAVIF = $(USE_LIBAVIF_$(TARGET_MAC_OS_X_VERSION_MAJOR));
-USE_LIBDAV1D_ENABLE_WEB_RTC_140000 = YES;
-USE_LIBDAV1D_ENABLE_WEB_RTC_150000 = YES;
+USE_LIBDAV1D[sdk=macos*] = $(USE_LIBDAV1D_$(ENABLE_WEB_RTC)$(WK_MACOS_1300));
+USE_LIBDAV1D[sdk=iphone*] = $(USE_LIBDAV1D_$(ENABLE_WEB_RTC)$(WK_IOS_16));
```
We don't support building for iOS 16 any more.

emw-apple committed:
[Xcode] WebCore fails to build for iOS when the macOS SDK is old
https://bugs.webkit.org/show_bug.cgi?id=264848
rdar://118356369

Reviewed by Alexey Proskuryakov.

Redefine USE_DAV1D with platform conditionals, so that iOS builds don't
depend on the version of the macOS SDK.

* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: https://commits.webkit.org/272438@main
